wardenclyffe/mediathread/views.py

```
@@ -124,70 +125,81 @@ def s3_upload(request):
     return HttpResponse("Bad file upload. Please try again.")
 
 
-@login_required
-@transaction.non_atomic_requests
-def video_mediathread_submit(request, id):
-    video = get_object_or_404(Video, id=id)
-    if request.method == "POST":
-        statsd.incr("mediathread.submit")
-        video.make_mediathread_submit_file(
-            video.filename(), request.user,
-            request.POST.get('course', ''),
-            redirect_to="",
-            audio=video.is_audio_file())
-        operations = video.handle_mediathread_submit()
-        for o in operations:
-            maintasks.process_operation.delay(o)
-        video.clear_mediathread_submit()
+def mediathread_url(username):
+    return (settings.MEDIATHREAD_BASE + "/api/user/courses?secret=" +
+            settings.MEDIATHREAD_SECRET + "&user=" +
+            username)
+
+
+class MediathreadCourseGetter(object):
+    def run(username):
+        try:
+            url = mediathread_url(username)
+            credentials = None
+            if hasattr(settings, "MEDIATHREAD_CREDENTIALS"):
+                credentials = settings.MEDIATHREAD_CREDENTIALS
+            response = GET(url, credentials=credentials)
+            courses = loads(response)['courses']
+            courses = [dict(id=k, title=v['title'])
+                       for (k, v) in courses.items()]
+            courses.sort(key=lambda x: x['title'].lower())
+        except:
```

any way to resolve the bare except?